packages/api/cms-api/src/generator/generate-crud.ts

```diff
@@ -450,12 +552,33 @@ function generateResolver({ generatorOptions, metadata }: { generatorOptions: Cr
             ${scopeProp ? `@Args("scope", { type: () => ${scopeProp.type} }) scope: ${scopeProp.type},` : ""}
             @Args("input", { type: () => ${classNameSingular}Input }) input: ${classNameSingular}Input
         ): Promise<${metadata.className}> {
-            const ${instanceNameSingular} = this.repository.create({
-                ...input,
-                ${blockProps.length ? `${blockProps.map((prop) => `${prop.name}: input.${prop.name}.transformToBlockData()`).join(", ")}, ` : ""}
-                ${hasVisibleProp ? `visible: false,` : ""}
-                ${scopeProp ? `scope,` : ""}
-            });
+            const { ${inputRelationToManyProps.map((prop) => `${prop.name}: ${prop.name}Input`).join(", ")}${
+        inputRelationToManyProps.length ? ", " : ""
+    }...assignInput } = input;
+                    const ${instanceNameSingular} = this.repository.create({
+                        ...assignInput,
+                        ${hasVisibleProp ? `visible: false,` : ""}
+                        ${scopeProp ? `scope,` : ""}        
+                        ${
+                            blockProps.length
+                                ? `${blockProps.map((prop) => `${prop.name}: input.${prop.name}.transformToBlockData()`).join(", ")}, `
+                                : ""
+                        }
+                        ${inputRelationManyToOneProps.map(
+                            (prop) =>
+                                `${prop.name}: ${prop.nullable ? `input.${prop.name} ? ` : ""}Reference.create(await this.${
+                                    prop.repositoryName
+                                }.findOneOrFail(input.${prop.name}))${prop.nullable ? ` : undefined` : ""}, `,
+                        )}
+                    });
+                    ${inputRelationToManyProps.map(
+                        (prop) => `{
+                        const ${prop.name} = await this.${prop.repositoryName}.find({ id: ${prop.name}Input });
+                        if (${prop.name}.length != ${prop.name}Input.length) throw new Error("Couldn't find all ${prop.name} that where passes as input");
+                        await ${instanceNameSingular}.${prop.name}.loadItems();
+                        ${instanceNameSingular}.${prop.name}.set(${prop.name}.map((p) => Reference.create(p)));
```

Comment:
Why "p"?

Reply:
p stands for product :) which is completely wrong of course